tools/dm-data-synchronization-features.md

```diff
@@ -0,0 +1,489 @@
+
+> **Note:**
+>
+> - Configuring multiple different routing rules for a single table is not supported.
+> - The match rule of schema needs to be configured separately, which is used to synchronize `create/drop schema xx`, as shown in `rule-2` of the [parameter configuration](#parameter-configuration).
+
+### Parameter configuration
+
+```yaml
+routes:
+  rule-1:
+    schema-pattern: "test_*"
+    table-pattern: "t_*"
+    target-schema: "test"
+    target-table: "t"
+  rule-2:
+    schema-pattern: "test_*"
+    target-schema: "test"
+```
+
+### Parameter explanation
+
+DM synchronizes the upstream MySQL or MariaDB instance table that matches the [`schema-pattern`/`table-pattern` rule provided by Table selector](#table-selector) to the downstream `target-schema`/`target-table`.
+
+### Usage examples
+
+This sections shows the usage examples in different scenarios.
+
+#### Merge sharded schemas and tables
+
+Assuming in the scenario of sharded schemas and tables, you want to synchronize two upstream MySQL instances `test_{1,2,3...}`.`t_{1,2,3...}` to the downstream TiDB instances `test`.`t`.
+
+To synchronize the upstream instances to the downstream `test`.`t`, you must create two routing rules:
+
+- `rule-1` is used to synchronize DML or DDL statements of the table that matches `schema-pattern: "test_*"` and `table-pattern: "t_*"` to the downstream `test`.`t`.
+- `rule-2` is used to synchronize DDL statements of the schema that matches `schema-pattern: "test_*"`, such as `create/drop schema xx`.
+
+> **Note:**
+>
+> - If the downstream `schema: test` already exists and will not be deleted, you can omit `rule-2`.
+> - If the downstream `schema: test` does not exist and only `rule-1` is configured, then it reports the `schema test doesn't exist` error during synchronization.
+
+```yaml
+  rule-1:
+    schema-pattern: "test_*"
+    table-pattern: "t_*"
+    target-schema: "test"
+    target-table: "t"
+  rule-2:
+    schema-pattern: "test_*"
+    target-schema: "test"
+```
+
+#### Merge sharded schemas
+
+Assuming in the scenario of sharded schemas, you want to synchronize two upstream MySQL instances `test_{1,2,3...}`.`t_{1,2,3...}` to the downstream TiDB instances `test`.`t_{1,2,3...}`.
+
+To synchronize the upstream schemas to the downstream `test`.`t_[1,2,3]`, you only need to create one routing rule.
+
+```yaml
+  rule-1:
+    schema-pattern: "test_*"
+    target-schema: "test"
+```
+
+#### Incorrect table routing
+
+Assuming that the following two routing rules are configured and `test_1_bak`.`t_1_bak` matches both `rule-1` and `rule-2`, an error is reported because the table routing configuration violates the number limitation.
+
+```yaml
+  rule-1:
+    schema-pattern: "test_*"
+    table-pattern: "t_*"
+    target-schema: "test"
+    target-table: "t"
+  rule-2:
+    schema-pattern: "test_1_bak"
+    table-pattern: "t_1_bak"
+    target-schema: "test"
+    target-table: "t_bak"
+```
+
+### Table selector
```

Comment:
*Table selector* is not a feature, but is used by some other features.
better structure of the doc?

Reply:
it is used by table routing, binlog event filtering, and column mapping.

Reply:
@csuzhangxc Got it! I'll put Table Selector into a separate document.